### ISSUE TITLE
Geting the result

### ISSUE THREAD

**Issue by maduperera** [journeyapps/zxing-android-embedded#218]
Hi,

Thank you for the wonderful Library and I was able to integrate it successfully.
I'm trying to run the barcode scanner in the same activity and get the result back to the same activity as below.

```
 @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.barcode_view);
        barcodeScannerView = (DecoratedBarcodeView)findViewById(R.id.zxing_barcode_scanner);
      
        capture = new CaptureManager(this, barcodeScannerView);
        capture.initializeFromIntent(getIntent(), savedInstanceState);
        capture.decode(); 
}
```

But it didn't work. I want to do something almost like the sample  "ToolBarCaptureActivity" you have provided. But in that sample we get the result back to the main activity.  Currently the CaptureManager has a method only to instantiate an activity from the result. 

```
protected void returnResult(BarcodeResult rawResult) {
        Intent intent = resultIntent(rawResult, getBarcodeImagePath(rawResult));
        activity.setResult(Activity.RESULT_OK, intent);
        finish();
}
```

In my case I need the result to get into the same activity.  
Can you please help me in this. 

![barcode](https://cloud.githubusercontent.com/assets/4883128/19161771/c3986cec-8c12-11e6-9976-5f7bacb75d70.png)

**Comment by rkistner** [comment]
Instead of using `CaptureManager#decode()`, you can directly use `barcodeScannerView.decodeSingle(callback)`.

**Comment by maduperera** [comment]
Thanks.  :) 
I tried the way you said just now, as below.

 ```
capture = new CaptureManager(this, barcodeScannerView);
 capture.initializeFromIntent(getIntent(), savedInstanceState);
 barcodeScannerView.decodeSingle(this);

```

I implemented the methods of the CallBack class in the current Activity so that I get the call backs to the very same activity. 

The camera view appears with the red line as expected. But it didn't detect the barcodes as before. And the yellow dots that represent the possible markers were not also displaying on the camera view. 

Why the barcodes didn't detected.
